meta-mender-commercial/recipes-mender/mender-monitor/mender-monitor_git.bb

```diff
@@ -1,16 +1,41 @@
 require mender-monitor.inc
 
-def mender_monitor_version_from_preferred_version(d):
+def mender_monitor_srcrev_from_src_uri(d, src_uri):
+    pref_version = d.getVar("PREFERRED_VERSION")
+    if pref_version is not None and pref_version.find("-build") >= 0:
+        # If "-build" is in the version, SRCREV won't be used for defining PV
+        return ""
+    else:
+        # Extract SRCREV from the tarball filename
+        import glob
+        import re
+        # Remove "file://" prefix
+        src_uri_glob = src_uri[len("file://"):]
```

Comment:
I would use `src_uri_glob = src_uri.split()[0][len("file://"):]`, otherwise multiple `SRC_URI` entries will confuse it. It requires that the tarball is the first entry, but that is more reasonable to assume.

Reply:
Good suggestion. Thanks